scriptmodules/emulators/duckstation.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="duckstation"
+rp_module_desc="PlayStation emulator DuckStation"
+rp_module_help="ROM Extensions: .pbp .cue .bin .chd .img .ecm .mds .mdf\n\nCopy your PlayStation roms to $romdir/psx\n\nCopy the required BIOS file to $biosdir"
+rp_module_licence="GPL3 https://raw.githubusercontent.com/stenzek/duckstation/master/LICENSE"
+rp_module_section="exp"
+rp_module_flags=""
+
+function depends_duckstation() {
+    local depends=(cmake libsdl2-dev libsnappy-dev pkg-config libevdev-dev libgbm-dev libdrm-dev)
+    getDepends "${depends[@]}"
+}
+
+function sources_duckstation() {
+    gitPullOrClone "$md_build" https://github.com/stenzek/duckstation.git
+}
+
+function build_duckstation() {
+    cmake -DCMAKE_BUILD_TYPE=Release -DENABLE_DISCORD_PRESENCE=OFF -DUSE_X11=OFF -DUSE_DRMKMS=ON -DBUILD_NOGUI_FRONTEND=ON -DBUILD_QT_FRONTEND=OFF .
```

Comment:
Shouldn't the `X11` support be enabled for `x11` platforms ? (this might need additional dependencies also).

Reply:
Yes it's needed X11 enabled option platform also will be great option enable option for Desktop QT/SDL2 BUILD frontend will be better for some custom option in the GUI.
And DRMKMS only for only KMS platform .

Reply:
i don't have an X11 platform to test but have enabled x11, and drmkms where appropriate